tests/bluetooth/bsim_bt/bsim_test_mesh_pst/src/settings_test_backend.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "settings_test_backend.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include "kernel.h"
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+
+#include <bluetooth/mesh.h>
+
+#define LOG_MODULE_NAME settings_test_backend
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define SETTINGS_FILE "/settings.log"
+#define ENTRY_LEN_SIZE (4)
+#define ENTRY_SEPERATOR ','
+#define ENTRY_NAME_MAX_LEN (SETTINGS_MAX_NAME_LEN + SETTINGS_EXTRA_LEN)
+#define ENTRY_VAL_MAX_LEN  (SETTINGS_MAX_VAL_LEN * 2)
+#define ENTRY_LINE_MAX_LEN (ENTRY_LEN_SIZE + ENTRY_NAME_MAX_LEN + ENTRY_VAL_MAX_LEN + sizeof(ENTRY_SEPERATOR) * 2)
+
+static char backend_file[255];
+static int fp;
+static bool backed_init;
+
+struct line_read_ctx {
+	off_t offset;
+	int len;
+};
+
+static int entry_check_and_seek(const char *name)
+{
+	if (lseek(fp, 0, SEEK_SET) < 0) {
+		return -1;
+	}
+
+	const char *namesv = name;
+	char c;
+
+	while (1) {
```

Comment:
Shouldn't it be until EOF to not get stuck forever loop?

Reply:
This is not an issue. Because the `read()` will make the loop to exit.